consensus/ibft/fork/manager.go

```diff
@@ -0,0 +1,434 @@
+package fork
+
+import (
+	"errors"
+	"path/filepath"
+
+	"github.com/0xPolygon/polygon-edge/consensus/ibft/hook"
+	"github.com/0xPolygon/polygon-edge/consensus/ibft/signer"
+	"github.com/0xPolygon/polygon-edge/secrets"
+	"github.com/0xPolygon/polygon-edge/validators"
+	"github.com/0xPolygon/polygon-edge/validators/store"
+	"github.com/0xPolygon/polygon-edge/validators/store/contract"
+	"github.com/0xPolygon/polygon-edge/validators/store/snapshot"
+	"github.com/hashicorp/go-hclog"
+)
+
+const (
+	loggerName                = "fork_manager"
+	snapshotMetadataFilename  = "metadata"
+	snapshotSnapshotsFilename = "snapshots"
+)
+
+var (
+	ErrForkNotFound           = errors.New("fork not found")
+	ErrSignerNotFound         = errors.New("signer not found")
+	ErrValidatorStoreNotFound = errors.New("validator set not found")
+)
+
+// ForkManager is an interface of the module that has Fork configuration and multiple version of submodules
+// and returns the proper submodule at specified height
+type ForkManager interface {
+	Initialize() error
+	GetSigner(uint64) (signer.Signer, error)
+	GetValidatorStore(uint64) (store.ValidatorStore, error)
+	GetValidators(uint64) (validators.Validators, error)
+	GetHooks(uint64) (hook.Hooks, error)
+	Close() error
+}
+
+// forkManagerImpl is a implementation of ForkManager
+type forkManagerImpl struct {
+	logger         hclog.Logger
+	blockchain     store.HeaderGetter
+	executor       contract.Executor
+	secretsManager secrets.SecretsManager
+
+	// configuration
+	forks     []IBFTFork
+	filePath  string
+	epochSize uint64
+
+	signers       map[validators.ValidatorType]signer.Signer
+	validatorSets map[store.SourceType]store.ValidatorStore
+}
```

Comment:
Interface and implementation in the same package.

Reply:
Explained in other comment

Reply:
Moved at b48e54ab25a49fb7d30c86df0773959f533ca652